File: Creational/Builder/src/applicationexample/builders/Builder.java
package applicationexample.builders;

import applicationexample.applicationcomponents.DataBase;
import applicationexample.applicationcomponents.OperationSystem;
import applicationexample.applicationcomponents.Type;
import applicationexample.applications.Application;

public interface Builder {

    void setVersion(double version);

    void setName(String name);

    void setDevelopersQuantity(int developersQuantity);

    void setOperationSystem(OperationSystem operationSystem);

    void setLanguage(String language);

    void setUserInterface(boolean userInterface);

    void setType(Type type);

    void setDataBase(DataBase dataBase);

    Application getResult();
}

File: Creational/Builder/src/singletonbuilder/builders/Builder.kt
package singletonbuilder.builders

import singletonbuilder.applications.Application

interface Builder {
    fun defineVersion(version: Double)
    fun defineName(name: String)
    fun defineDevelopersQuantity(developersQuantity: Int)
    fun defineOperatingSystem(operatingSystem: String)
    fun defineLanguage(language: String)
    fun getResult(): Application
}